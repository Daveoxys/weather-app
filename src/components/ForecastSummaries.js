import React from "react";
import PropTypes from "prop-types";
import ForecastSummary from "./ForecastSummary";

const ForeCastSummaries = ({ forecasts }) => (
  <div className="forecast-summaries">
    {forecasts.map((forecast) => (
      <ForecastSummary
        date={forecast.date}
        description={forecast.description}
        icon={forecast.icon}
        temperature={forecast.temperature}
      />
    ))}
  </div>
);

ForeCastSummaries.propTypes = {
  forecasts: PropTypes.string.isRequired,
};

export default ForeCastSummaries;
