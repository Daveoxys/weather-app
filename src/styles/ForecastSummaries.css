.forecast-summaries {
  display: flex;
  justify-content: space-between;
  text-align: center;
}
