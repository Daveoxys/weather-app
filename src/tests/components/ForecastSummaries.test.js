import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 1111111,
      description: "Stub description",
      icon: "800",
      temperature: {
        max: 22,
        min: 12,
      },
    },
    {
      date: 2222222,
      description: "Stub description2",
      icon: "602",
      temperature: {
        max: 24,
        min: 13,
      },
    },
  ];
  it("renders correctly", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={validProps} />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
