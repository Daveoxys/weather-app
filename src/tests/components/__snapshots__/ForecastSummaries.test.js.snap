// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
    div-testid="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        <h3>
          Thu 1st Jan
        </h3>
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description
      </div>
      <button
        type="button"
      >
        More Details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        <h3>
          Thu 1st Jan
        </h3>
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-snow"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        24°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description2
      </div>
      <button
        type="button"
      >
        More Details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
